src/UI/Implementation/Component/Input/Field/FilterContextRenderer.php

```diff
@@ -0,0 +1,293 @@
+
+		if ($input instanceof Component\Input\Field\Text) {
+			$input_tpl = $this->getTemplate("tpl.text.html", true, true);
+		} elseif ($input instanceof Component\Input\Field\Select) {
+			$input_tpl = $this->getTemplate("tpl.select.html", true, true);
+		} else {
+			throw new \LogicException("Cannot render '" . get_class($input) . "'");
+		}
+
+		return $this->renderProxyFieldWithContext($input_tpl, $input, $default_renderer);
+	}
+
+
+	/**
+	 * @param Group             $group
+	 * @param RendererInterface $default_renderer
+	 *
+	 * @return string
+	 */
+	protected function renderFieldGroups(Group $group, RendererInterface $default_renderer) {
+
+		$inputs = "";
+		$input_labels = array();
+		foreach ($group->getInputs() as $input) {
+			$input = $input->withByLine($group->getLabel());
+			$inputs .= $default_renderer->render($input);
+			$input_labels[] = $input->getLabel();
+		}
+		if ($group->getLabel() != "disabled")
+		{
+			$inputs .= $this->renderAddField($input_labels, $default_renderer);
+		}
+
+		return $inputs;
+	}
+
+
+	/**
+	 * @param Template $input_tpl
+	 * @param Input    $input
+	 * @param RendererInterface $default_renderer
+	 *
+	 * @return string
+	 */
+	protected function renderProxyFieldWithContext(Template $input_tpl, Input $input, RendererInterface $default_renderer) {
+
+		$f = $this->getUIFactory();
+		$tpl = $this->getTemplate("tpl.context_filter.html", true, true);
+
+		if ($input->getByline() == "disabled") {
+			$remove_glyph = $f->glyph()->remove()->withUnavailableAction(true);
+		} else {
+			$remove_glyph = $f->glyph()->remove()->withAdditionalOnLoadCode(function ($id) {
+				$code = "$('#$id').on('click', function(event) {
+							il.UI.filter.onRemoveClick(event, '$id');
+							return false; // stop event propagation
+					});";
+				//var_dump($code); exit;
+				return $code;
+			});
+		}
+
+		$tpl->setCurrentBlock("addon_left");
+		$tpl->setVariable("LABEL", $input->getLabel());
+		$tpl->parseCurrentBlock();
+		$tpl->setCurrentBlock("filter_field");
+		$tpl->setVariable("FILTER_FIELD", $this->renderProxyField($input_tpl, $input, $default_renderer));
+		$tpl->parseCurrentBlock();
+		$tpl->setCurrentBlock("addon_right");
+		$tpl->setVariable("DELETE", $default_renderer->render($remove_glyph));
+		$tpl->parseCurrentBlock();
+
+		return $tpl->get();
+	}
+
+
+	/**
+	 * @param Template $tpl
+	 * @param Input    $input
+	 * @param RendererInterface    $default_renderer
+	 *
+	 * @return string
+	 */
+	protected function renderProxyField(Template $input_tpl, Input $input, RendererInterface $default_renderer) {
+
+		$f = $this->getUIFactory();
+		$tpl = $this->getTemplate("tpl.filter_field.html", true, true);
+
+		$content = $this->renderInputFieldWithContext($input_tpl, $input);
+		$popover = $f->popover()->standard($f->legacy($content))->withVerticalPosition();
+		$tpl->setVariable("POPOVER", $default_renderer->render($popover));
+
+		$prox = new ProxyFilterField();
+		if ($input->getByline() != "disabled")
```

Comment:
Yes, exactly. Unfortunately we do not have disabled inputs yet. I think we need this for forms as well in the future. I would mark this as a workaround since there is no other way to get this info through (as far as I know). The bylines are currently not needed. If there is currently a better way to achieve this (with the current interfaces) I would be happy do change this.